# Iris
**Author: A. O'Connor**
**************
This notebook contains my own analysis of the Iris dataset. Iris is a famous data set available on [Github](https://gist.github.com/curran/a08a1080b88344b0c8a7). It will be loaded into this notebook for data visualisation and exploration. Each row of the table represents an iris flower from one of three species of iris.  
**************

### Import Libraries
The Pandas library will be used for interpretation and analysis of the data set.\
The Matplotlib pyplot library will be used for plotting the data.
*************

In [1]:
# Install pandas and matplotlib
import pandas as pd
import matplotlib.pyplot as plt

### Load the data set
The Iris data set will be loaded from [GitHub]("https://gist.githubusercontent.com/curran/a08a1080b88344b0c8a7/raw/0e7a9b0a5d22642a06d3d5b9bcbad9890c8ee534/iris.csv"), using Pandas to read in the raw csv file. 
***************

In [3]:
# Load the dataset using pandas read_csv() function
df = pd.read_csv("https://gist.githubusercontent.com/curran/a08a1080b88344b0c8a7/raw/0e7a9b0a5d22642a06d3d5b9bcbad9890c8ee534/iris.csv")

### Overview of the data
Initially, view the the data frame and the summary statistics of the overall data set, to get an overall idea of the data.
**********

In [4]:
# View the data frame
df

sepal_length  sepal_width  petal_length  petal_width    species
0             5.1          3.5           1.4          0.2     setosa
1             4.9          3.0           1.4          0.2     setosa
2             4.7          3.2           1.3          0.2     setosa
3             4.6          3.1           1.5          0.2     setosa
4             5.0          3.6           1.4          0.2     setosa
..            ...          ...           ...          ...        ...
145           6.7          3.0           5.2          2.3  virginica
146           6.3          2.5           5.0          1.9  virginica
147           6.5          3.0           5.2          2.0  virginica
148           6.2          3.4           5.4          2.3  virginica
149           5.9          3.0           5.1          1.8  virginica

[150 rows x 5 columns]

In [5]:
# Summary statistics of the data frame 
df.describe()

sepal_length  sepal_width  petal_length  petal_width
count    150.000000   150.000000    150.000000   150.000000
mean       5.843333     3.054000      3.758667     1.198667
std        0.828066     0.433594      1.764420     0.763161
min        4.300000     2.000000      1.000000     0.100000
25%        5.100000     2.800000      1.600000     0.300000
50%        5.800000     3.000000      4.350000     1.300000
75%        6.400000     3.300000      5.100000     1.800000
max        7.900000     4.400000      6.900000     2.500000

In [7]:
# Information about the data frame 
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   sepal_length  150 non-null    float64
 1   sepal_width   150 non-null    float64
 2   petal_length  150 non-null    float64
 3   petal_width   150 non-null    float64
 4   species       150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB


#### Information about the data set
*******
The Iris data set contains 150 rows and 5 columns.\
These column contain various data types. 
- Variables in the columns sepal length, sepal width, petal length and petal width are all numeric with multiple decimal places, therefore they  are the float data type. They are stored in the Pandas data frame as DType Float64. 
- Variables in the column species are strings (i.e. species can be "setosa", "versicolor" or "virginica".). In Pandas data frames, strings are by default stored as the object data type. 

No values are missing in the columns. 

## Correlations
****
Using Pandas to view the correlations in the data.\
Pandas returns a correlation matrix of numeric columns.\
From a Pandas tutorial on [W3 Schools.](https://www.w3schools.com/python/pandas/pandas_correlations.asp)
**************

In [9]:
#Create a numeric data frame from the overall data frame using the function select_dtypes and specifying to include numeric columns. 
#Reference for this: https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.select_dtypes.html
df_numeric = df.select_dtypes(include = ('number'))

#Use the function .corr() from the Pandas library to view the correlations in the numeric data frame
df_numeric.corr()

sepal_length  sepal_width  petal_length  petal_width
sepal_length      1.000000    -0.109369      0.871754     0.817954
sepal_width      -0.109369     1.000000     -0.420516    -0.356544
petal_length      0.871754    -0.420516      1.000000     0.962757
petal_width       0.817954    -0.356544      0.962757     1.000000

This correlation matrix describes the relationship between each numeric column in the data frame, represented as an R value ranging from -1 to +1.\
Negative numbers represent a negative correlation between the two variables (i.e. as one number increases the other decreases), and positive numbers represent a positive correlation. The closer the R value is to zero, the less strong the correlation. In the Iris data set, the following correlations are worth noting:
- Sepal length is positively correlated with petal length, R value of 0.87.
- Sepal length is also positively correlated with petal width, R value of 0.82.
- The strongest correlation exists between petal length and petal width, with an R value of 0.96. 